pkg/cache/v2/cache.go

```diff
@@ -40,32 +41,97 @@ type ConfigWatcher interface {
 	//
 	// Cancel is an optional function to release resources in the producer. If
 	// provided, the consumer may call this function multiple times.
-	CreateWatch(Request) (value chan Response, cancel func())
+	CreateWatch(Request) (value chan ResponseIface, cancel func())
```

Comment:
After more thought, what do you think of keeping this named `Response`, but changing the `Response` struct to be named `DeconstructedResponse` or something? I just think passing `Response` around looks cleaner than `ResponseIface`, but I don't feel too strongly about this.

Reply:
I think I like ResponseIFace because it represents an interface.